lib/client.js

```diff
@@ -129,9 +129,12 @@ function MqttClient (streamBuilder, options) {
   // Inflight callbacks
   this.outgoing = {}
 
+  this.connectEmitter = null
+
   // Mark connected on connect
-  this.on('connect', function () {
+  this._onConnect = function () {
```

Comment:
can you move this to a top level function instead?

Reply:
Yes, I will do that.